tpch/queries/q16.py

```diff
@@ -0,0 +1,26 @@
+import narwhals as nw
+from narwhals.typing import FrameT
+
+
+@nw.narwhalify
+def query(part_ds: FrameT, partsupp_ds: FrameT, supplier_ds: FrameT) -> FrameT:
+    var1 = "Brand#45"
+
+    supplier = supplier_ds.filter(
+        nw.col("s_comment").str.contains(".*Customer.*Complaints.*")
+    ).select(nw.col("s_suppkey"), nw.col("s_suppkey").alias("ps_suppkey"))
+
+    return (
+        part_ds.join(partsupp_ds, left_on="p_partkey", right_on="ps_partkey")
+        .filter(nw.col("p_brand") != var1)
+        .filter(~nw.col("p_type").str.contains("MEDIUM POLISHED*"))
+        .filter(nw.col("p_size").is_in([49, 14, 23, 45, 19, 3, 36, 9]))
+        .join(supplier, left_on="ps_suppkey", right_on="s_suppkey", how="left")
+        .filter(nw.col("ps_suppkey_right").is_null())
+        .group_by("p_brand", "p_type", "p_size")
+        .agg(nw.col("ps_suppkey").unique().len().alias("supplier_cnt"))
```

Comment:
`unique().len()` changed from `n_unique()`

Because `AttributeError: 'SeriesGroupBy' object has no attribute 'n_unique'.`

Reply:
thanks @luke396 ! looks like a bug, will try to resolve beforehand https://github.com/narwhals-dev/narwhals/issues/915

Reply:
I made an attempt in #917, I don't have time to test against the queries right now though 🙈